lib/genome/groupers/interaction_grouper.rb

```diff
@@ -1,16 +1,19 @@
 module Genome
   module Groupers
     class InteractionGrouper
+      attr_reader :group_from_scratch
```

Comment:
This will create an instance level accessor for `group_from_scratch` but all the methods here are class methods (denoted by the `def self.method_name` so it will not be usable in that context. You can access `@group_from_scratch` directly or you could add the accessor to the metaclass

```
class << self
  attr_reader :group_from_scratch
end
```

or if that's all too weird we could just pass the value along from `run()` to each of the sub-methods or switch things to be normal instance methods and `new` up a grouper.

I think I prefer one of those last two options myself. I generally make things just static class methods when there's no internal state, but if we want to start storing internal state for a class it makes more sense to either not store the state and just propagate it along, or convert to a normal class with instance level methods and variables.

I think actually it will "work" as written because you only access `@run_from_scratch` later in the code rather than through the accessor method, but if someone tried to use that accessor it would break.

Reply:
I changed this to pass along `group_from_scratch` to the sub-methods.